CRM/Campaign/Selector/Search.php

```diff
@@ -296,7 +296,7 @@ public function buildPrevNextCache($sort) {
 
       if (Civi::service('prevnext') instanceof CRM_Core_PrevNextCache_Sql) {
         // SQL-backed prevnext cache uses an extra record for pruning the cache.
-        CRM_Core_BAO_Cache::setItem($cacheKey, 'CiviCRM Search PrevNextCache', $cacheKey);
+        Civi::cache('prevNextCache')->set($cacheKey, $cacheKey);
```

Comment:
@totten agreed that the effect of 
```
Civi::cache('prevNextCache')->set($cacheKey, $cacheKey);
```
is a clumsy way of saying
```

Civi::cache('prevNextCache')->set($cacheKey,  TRUE);
```

Reply:
Just to be clear - I support setting the value in the same way it was set before. But Eileen had a question about whether it was meaningful to do `set($cacheKey,$cacheKey)`, and (conceptually) these could be the same (although I'd make no assumption that they're actually equivalent in practice).

Also, note that `CRM_Core_BAO_PrevNextCache::cleanupCache()` specifically relies on this cache being stored in SQL. I'd recommend skeptical eye about using a cache-backend that is dynamically-chosen (as in `prevNextCache' => 'CiviCRM Search PrevNextCache' ... 'type' => ['*memory*', 'SqlGroup', 'ArrayCache']` in `Container.php` below) when there are hard-coded bits that assume this cache is SQL-backed...

Reply:
@totten good point have just pushed up change to ensure prevNextCache is sql Backed

Reply:
@totten should `CRM_Core_BAO_PrevNextCache:::cleanupCache()` be aware of the backed provider of prev next service?